test/framework/clustervalidator.go

```diff
@@ -109,79 +145,84 @@ func validateEKSAObjects(ctx context.Context, validateOpts clusterOpts) error {
 			key.Namespace = "default"
 		}
 
-		if err := clusterClient.Get(ctx, key, u); err != nil {
+		if err := mgmtClusterClient.Get(ctx, key, u); err != nil {
 			return fmt.Errorf("cluster object does not exist %s", err)
 		}
 	}
-
+	logger.V(4).Info("EKSA objects exists validated", "cluster", clus.Name)
 	return nil
 }
 
-func validateClusterReady(ctx context.Context, validateOpts clusterOpts) error {
-	clusterClient := validateOpts.ClusterClient
-	if validateOpts.ClusterSpec.Cluster.IsManaged() {
-		clusterClient = validateOpts.ManagementClusterClient
-	}
-
-	capiCluster, err := controller.GetCAPICluster(ctx, clusterClient, validateOpts.ClusterSpec.Cluster)
+func validateClusterReady(ctx context.Context, vc ClusterValidatorConfig) error {
+	clus := vc.ClusterSpec.Cluster
+	mgmtClusterClient := vc.GetManagementClusterClient()
+	capiCluster, err := controller.GetCAPICluster(ctx, mgmtClusterClient, clus)
 	if err != nil {
 		return fmt.Errorf("failed to retrieve cluster %s", err)
 	}
-
 	if capiCluster == nil {
-		return fmt.Errorf("cluster %s does not exist", validateOpts.ClusterSpec.Cluster.Name)
+		return fmt.Errorf("cluster %s does not exist", clus.Name)
 	}
-
-	if err := checkClusterReady(capiCluster); err != nil {
```

Comment:
Maybe it's because it's Monday, but I don't see the difference between this and what replaced it. Is there a reason for the replacement?

Reply:
There is no difference really. I got rid of the function wrapping the logic entirely because it wasn't being used anywhere else.